### PR TITLE
Solution and Unit test for Problems set by Khoa.

### PR DESCRIPTION
**Pull request by BrianNguyen420**

### CODE REVIEW

File: simpleSequential/isEmptySet.py [hunk 1]
```diff
@@ -3,3 +3,5 @@
 # Ex:
 #   isEmptySet({})     # Return True
 #   isEmptySet({2, 3}) # Return False
+def isEmptySet(oneSet: set) -> bool:
```
Comment (rubikvn2100) [inline simpleSequential/isEmptySet.py]: 1/ Let use's "intSet" instead of "oneSet". Same the amount of characters but more meaning, and cover the actual type which is a "set of integers".

2a/ Did you import "Set" from typing?

2b/ Please specify this type as a "set of integers".

File: simpleSequential/isEmptySet.py [hunk 2]
```diff
@@ -3,3 +3,5 @@
...
+    return  len(oneSet) == 0
```
Comment (rubikvn2100) [inline simpleSequential/isEmptySet.py]: typo, have extra space after the return keyword.

File: simpleSequential/isPerfectSquare.py [hunk 3]
```diff
@@ -11,4 +11,6 @@
 # Hint: This problem can be done by combine multiple simple operators.
 # Ex:
 #   isPerfectSquare(16) # return True
-#   isPerfectSquare(17) # return False
```
Comment (rubikvn2100) [inline simpleSequential/isPerfectSquare.py:14]: You should go back and figure out why this is the case. Prevent the "change" in unnecessary lines. You never know what you change there. Especially when it is not oblivious here.

Tips: Copy your code block in lines 15 and 16. In VS code, click reset the code block. Then past your own code back.

File: simpleSequential/isPerfectSquare.py [hunk 4]
```diff
@@ -11,4 +11,6 @@
...
\ No newline at end of file
+#   isPerfectSquare(17) # return False
+def isPerfectSquare(n: int) -> bool:
+   return n >= 0 and int(n**0.5)**2 == n
```
Comment (rubikvn2100) [inline simpleSequential/isPerfectSquare.py]: 1/ Look like your indentation is wrong here. The other files have an indent of 4 characters. I have a feeling that you write this line of code without running the unit test again.

2/ Typos, needs space around operation. Something like "n1 ** n2" not "n1**n2".

File: simpleSequential/isTwoSetEqual.py [hunk 5]
```diff
@@ -11,3 +11,5 @@
 #   isTwoSetEqual({2, 3}, {2, 3, 2, 2, 3}) # Return True
 #   isTwoSetEqual({2, 3}, {2, 3, 1, 5, 4}) # Return False
 #   isTwoSetEqual({2, 3}, {123, 456, 789}) # Return False
+def isTwoSetEqual(firstSet: set, secondSet: set) -> bool:
```
Comment (rubikvn2100) [inline simpleSequential/isTwoSetEqual.py]: 1/ Let use's "intSet1" and "intSet2" instead of "firstSet" and "secondSet". The amount of characters is less but more meaningful, and covers the actual type which is a "set of integers".

2a/ Did you import "Set" from typing?

2b/ Please specify this type as a "set of integers".

File: simpleSequential/isTwoSetEqual.py [hunk 6]
```diff
@@ -11,3 +11,5 @@
...
+        return firstSet == secondSet
```
Comment (rubikvn2100) [inline simpleSequential/isTwoSetEqual.py]: The indentation here looks wrong. Look like you indent two times (two tabs?).

File: simpleSequential/test_floatToInteger.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import unittest
+from floatToInteger import floatToInteger
+
+class TestFloatToInteger(unittest.TestCase):
+    def test_convertFloatToInteger(self):
```
Comment (rubikvn2100) [inline simpleSequential/test_floatToInteger.py:5]: No mistake here. Just want to say that the test description is clear.

File: simpleSequential/test_isEmptySet.py [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import unittest
+from isEmptySet import isEmptySet
+
+class TestIsEmptySet(unittest.TestCase):
+    def test_emptyInput(self):
```
Comment (rubikvn2100) [inline simpleSequential/test_isEmptySet.py]: What is "empty"? string, dictionary, list, or set? Let use "emptyIntegerSetInput" instead of "emptyInput".

Note: A test description needs to be short, but needs to be meaningful and clear.

File: simpleSequential/test_isEmptySet.py [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+        self.assertEqual(isEmptySet({}), True)
+
+    def test_setIntegerInput(self):
```
Comment (rubikvn2100) [inline simpleSequential/test_isEmptySet.py]: How about "nonEmptySetInput"? It is still meaningful and shows the contrast with the other test. And we are focusing on the "empty" or "nonEmpty" in this problem remember?

Note: A unit test need to focus on testing the functionality of a code. So this code is about empty or nonempty, so better to write descriptions around that.

File: simpleSequential/test_isPerfectSquare.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import unittest
+from isPerfectSquare import isPerfectSquare
+
+class TestIsPerfectSquare(unittest.TestCase):
```
Comment (rubikvn2100) [inline simpleSequential/test_isPerfectSquare.py:4]: Great, but one small change to make it better. Let's put those tests that are close together in meaning. So that it is easier to review. My recommendation of the order:

1/ Perfect Square Integer Input
2/ Non Perfect Square Integer Input 
3/ Negative Input
4/ Number Zero Input

If you look at the order above, you will see

Group 1 (test 1 and test 2): main functionality of the function.
Group 2 (test 3): the special case.
Group 3 (test 4): the edge case.

File: simpleSequential/test_isTwoSetEqual.py [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import unittest
+from isTwoSetEqual import isTwoSetEqual
+
+class TestIsTwoSetEqual(unittest.TestCase):
```
Comment (rubikvn2100) [inline simpleSequential/test_isTwoSetEqual.py:4]: Yeah, but I recommend those 2 tests with names like:

1/ Equal Integer Set Input.
2/ Non Equal Integer Set Input.

At least you need to change "Set Integer" to "Integer Set" for it to be Grammatically correct. This is really important for future recruiters to look at.

File: simpleSequential/test_isTwoSetEqual.py [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+    def test_equalSetIntegerInput(self):
+        self.assertEqual(isTwoSetEqual({6, 9}, {9, 6}), True)
+
+    def test_notEqualSetIntegerInput(self):
```
Comment (rubikvn2100) [inline simpleSequential/test_isTwoSetEqual.py]: "non" or "not"? you are not consistence in different files. Sometimes you use "nonEqual", sometime you use "notEqual".

I think you should stick to "non". Fix this.

File: simpleSequential/test_unionTwoSet.py [hunk 13]
```diff
@@ -0,0 +1,12 @@
+import unittest
+from unionTwoSet import unionTwoSet
+
+class TestUnionTwoSet(unittest.TestCase):
+    def test_equalSetIntegerInput(self):
+        self.assertEqual(unionTwoSet({2, 3}, {3, 2}), {2, 3})
+
+    def test_noneEqualSetIntegerInput(self):
```
Comment (rubikvn2100) [inline simpleSequential/test_unionTwoSet.py]: "non" or "none"? you are not consistence in different files. Sometimes you use "nonEqual", sometime you use "noneEqual".

I think you should stick to "non". Fix this.

File: simpleSequential/test_unionTwoSet.py [hunk 14]
```diff
@@ -0,0 +1,12 @@
+import unittest
+from unionTwoSet import unionTwoSet
+
+class TestUnionTwoSet(unittest.TestCase):
```
Comment (rubikvn2100) [inline simpleSequential/test_unionTwoSet.py:4]: At least you need to change "Set Integer" to "Integer Set" for it to be Grammatically correct. This is really important for future recruiters to look at.

File: simpleSequential/test_isEmptySet.py [hunk 15]
```diff
@@ -0,0 +1,12 @@
+import unittest
+from isEmptySet import isEmptySet
+
+class TestIsEmptySet(unittest.TestCase):
```
Comment (rubikvn2100) [inline simpleSequential/test_isEmptySet.py:4]: Will it return False when the set is not empty? How do you know if you not test it?

By the way. Here is my recommend name:

1/ emptyIntegerSetInput
2/ nonEmptyIntegerSetInput

File: simpleSequential/unionTwoSet.py [hunk 16]
```diff
@@ -3,3 +3,5 @@
 #   unionTwoSet({2, 3}, {3, 2}) # Return a set that equal to {2, 3}
 #   unionTwoSet({2, 3}, {4, 3}) # Return a set that equal to {2, 3, 4}
 #   unionTwoSet({2, 3}, {5, 4}) # Return a set that equal to {2, 3, 4, 5}
+def unionTwoSet(firstSet: set, secondSet: set) -> set:
```
Comment (rubikvn2100) [inline simpleSequential/unionTwoSet.py]: 1/ Let use's "intSet1" and "intSet2" instead of "firstSet" and "secondSet". The amount of characters is less but more meaningful, and covers the actual type which is a "set of integers".

2a/ Did you import "Set" from typing?

2b/ Please specify this type as a "set of integers".

**Review by rubikvn2100 — CHANGES_REQUESTED** [review]
The review is mostly about

1/ Typo
2/ Missing import of Set type and Set[int] definition.
3/ Miss typing of indentation
4/ Inconsistent of "non", "not", and "none" in test description. Recommend using "non".
5/ Unclear variable/parameter name. What is "oneSet", "firstSet", or "secondSet"? How about "intSet"?
6/ Un expected line change in isPerfectSquare.py
7/ No spacing for operation in isPerfectSquare.py
8/ Grammatically wrong of using "setInteger" let use "integerSet" remember "adj + N"
9/ Test ordering and grouping is important in test_isPerfectSquare.py file. Think about a test suite of 100 tests and you put them randomly instead of grouping them?

File: simpleSequential/isEmptySet.py (hunk 1 above)
Comment (BrianNguyen420) [reply simpleSequential/isEmptySet.py]: Acknowledge.

File: simpleSequential/isTwoSetEqual.py [hunk 17]
```diff
@@ -11,3 +11,7 @@
 #   isTwoSetEqual({2, 3}, {2, 3, 2, 2, 3}) # Return True
 #   isTwoSetEqual({2, 3}, {2, 3, 1, 5, 4}) # Return False
 #   isTwoSetEqual({2, 3}, {123, 456, 789}) # Return False
+from typing import Set
+
+def isTwoSetEqual(intSet1: Set[int], inSet2: Set[int]) -> bool:
```
Comment (rubikvn2100) [inline simpleSequential/isTwoSetEqual.py]: typo

File: simpleSequential/test_isPerfectSquare.py (hunk 10 above)
Comment (rubikvn2100) [reply simpleSequential/test_isPerfectSquare.py:4]: Did you check this one?

File: simpleSequential/test_isTwoSetEqual.py (hunk 11 above)
Comment (rubikvn2100) [reply simpleSequential/test_isTwoSetEqual.py:4]: This you check this one?

File: simpleSequential/unionTwoSet.py [hunk 18]
```diff
@@ -3,3 +3,7 @@
 #   unionTwoSet({2, 3}, {3, 2}) # Return a set that equal to {2, 3}
 #   unionTwoSet({2, 3}, {4, 3}) # Return a set that equal to {2, 3, 4}
 #   unionTwoSet({2, 3}, {5, 4}) # Return a set that equal to {2, 3, 4, 5}
+from typing import Set
+
+def unionTwoSet(intSet1: Set[int], intSet2: Set[int]) -> set:
```
Comment (rubikvn2100) [inline simpleSequential/unionTwoSet.py]: Wrong return type.
